web/api/query.go

```diff
@@ -46,9 +47,9 @@ func httpJSONError(w http.ResponseWriter, err error, code int) {
 	fmt.Fprintln(w, ast.ErrorToJSON(err))
 }
 
-func parseTimestampOrNow(t string) (clientmodel.Timestamp, error) {
+func parseTimestampOrNow(t string, nower utility.Time) (clientmodel.Timestamp, error) {
```

Comment:
Where would that apply, people writing `parseTimestampOrNow(timestamp, nil)` instead of `parseTimestampOrNow(timestamp, time.Now)`? The latter seems to be easier to understand for readers of the code base. No strong feelings either, but in general I'd favor the simpler solution.